### PR TITLE
Update project to AndroidX

### PR DESCRIPTION
**Pull request by lordcodes**
- Update to use AndroidX instead of the support library.
- Wear sample needed to use AndroidX Wear which is minimum API 23, however, this is only for the sample, it doesn't affect the actual wear library dependency.
- Update test/sample dependencies to latest versions.
- Update to latest Gradle version, Android Studio 3.6 Gradle plugin and compile using Android API 29.

Samples tested, all tests ran locally. Circle CI checks don't seem to appearing in the PR, so could a project maintainer let me know if I need to do anything for that.

There is already another PR (#618) to update to AndroidX, however, it has gone stale for a while.

### CODE REVIEW

File: analytics-samples/analytics-sample/src/main/java/com/segment/analytics/sample/SampleApp.java [hunk 1]
```diff
@@ -37,10 +39,14 @@
   public void onCreate() {
     super.onCreate();
 
-    CalligraphyConfig.initDefault(
-        new CalligraphyConfig.Builder()
-            .setDefaultFontPath("fonts/CircularStd-Book.otf")
-            .setFontAttrId(R.attr.fontPath)
+    ViewPump.init(
```
Comment (chrisjenx) [inline analytics-samples/analytics-sample/src/main/java/com/segment/analytics/sample/SampleApp.java:42]: Oh cool, didn't know you still use this 👍

**Comment by bsneed** [comment]
Wow @lordcodes; thanks so much for this!!!!

**Comment by chrisjenx** [comment]
awesome thanks @lordcodes!

**Comment by lordcodes** [comment]
@bsneed I mentioned in the PR description that it didn't look like the PR was running CI properly. It looks liked Circle CI only ran after you merged the PR, which failed due to an issue with test coverage.

Is having test coverage enabled crucial for you? It looks like there are issues with it, I have tried investigating why. It looks like the JaCoCo plugin the project uses has a task to generate the report that is failing:

```
Execution failed for task ':analytics:createDebugAndroidTestCoverageReport'.
> java.io.IOException: No coverage data to process in directories [PROJECT_DIR/analytics/build/outputs/code_coverage/debugAndroidTest/connected]
```

**Pull request merged**